src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -290,6 +290,29 @@ Transactor::checkSeqProxy(
             return tefNO_TICKET;
         }
     }
+    // Only a few transaction types reference the sequence of another
+    // transaction, but they all use the same fields.
+    if (auto const referenceSequence = tx[~sfOfferSequence])
+    {
+        // If the transaction uses the "Owner" field use that, else use the
+        // current account. Validation of those fields is done elsewhere.
+        auto const owner = tx[~sfOwner].value_or(id);
+        if (view.exists(keylet::ticket(owner, *referenceSequence)))
+        {
+            // Ticket execution order is non-deterministic. The owner account
+            // may have a transaction pending which consumes the existing ticket
+            // to create an object that this transaction depends on. Because the
+            // consequences of executing the two transactions in a different
+            // order can be significant, don't allow this one to be processed
+            // until the ticket has been consumed. This is analagous to
+            // "terPRE_SEQ" for tickets.
+            JLOG(j.debug())
+                << "Can not cancel offer " << *referenceSequence
+                << " because the corresponding ticket hasn't been used yet, "
+                << "and the order of Ticket processing can not be guaranteed.";
+            return terTICKET;
+        }
+    }
```

Comment:
I didn't consider the performance implications, so thank you for checking that.

I like your solution, but I am wide open to a discussion of whether the "problem" is a problem worth solving with either option. If we decide to keep this check, your commit is definitely the way to go. If not, it's super easy to remove this check.

In "real world" terms, I predict that it will be pretty unlikely that any use case that submits and cancels offers so quickly that this could be an issue is probably going to be a bot, and I think a bot is really unlikely to use tickets. And most escrows last much longer than a couple of consensus rounds, AND an escrow cancel or finish that uses a "bad" sequence will get a `tecNO_TARGET`, which is a perfectly acceptable result IMHO.

Reply:
That's kinda my feeling as well.  Agree with your assessment about how frequently this check will pay off.  There are definitely situations where the check can fire, but they seem low likelihood.  And if the check is removed the consequences seem acceptable.  My inclination is to remove the check altogether.  But I'm open to further discussion.